### DISCUSSION TITLE
QR Code not generating all characters via API call.

### DISCUSSION THREAD

**Discussion by barrett316** [metafloor/bwip-js#309] [Q&A] [answered]
I'm sure this is something on the input side, but when generating a QR code via an API call for example "jhfg!#@$?" , the result is "jhfg!" instead of the rest of the characters. Please advise on what I'm doing incorrectly.

**Comment by metafloor** [comment, marked as answer]
Considering that `#`, `@`, and `?` are special to URLs, my guess is you are not using `encodeURIComponent()` on the text value...

**Reply by barrett316** [reply]
I think you're correct. Can you give me an example of how the URL should look using the encodeURIComponent please.

Edit: I think I got it, thanks.

**Reply by metafloor** [reply]
You need to use percent-encoding - basically a percent sign `%` followed by the two-hexadecimal byte value.  For example, '#' will be encoded as `%23`, `%` as `%25`.  See: [Percent Encoding](en.wikipedia.org/wiki/Percent-encoding).
